portal-backend/tests/depmap/compound/test_legacy_utils.py

```diff
@@ -73,6 +74,81 @@ def test_get_compound_labels_by_experiment_labels(empty_db_mock_downloads):
     interactive_test_utils.reload_interactive_config()
 
     result = get_compound_labels_by_experiment_label(dataset_id)
-    assert list(result.keys()) == [compound_experiment_1A.label, compound_experiment_2.label]
-    assert result[compound_experiment_1A.label] == compound1.label # pyright: ignore
-    assert result[compound_experiment_2.label] == compound2.label # pyright: ignore
\ No newline at end of file
+    assert list(result.keys()) == [
+        compound_experiment_1A.label,
+        compound_experiment_2.label,
+    ]
+    assert result[compound_experiment_1A.label] == compound1.label  # pyright: ignore
+    assert result[compound_experiment_2.label] == compound2.label  # pyright: ignore
+
+
+def test_does_legacy_dataset_exist_with_compound_experiment(empty_db_mock_downloads):
+    dataset_name = DependencyDataset.DependencyEnum.Prism_oncology_AUC
+    dataset_name_repurposing = DependencyDataset.DependencyEnum.Rep_all_single_pt
+
+    # Compound 1 has 3 corresponding experiments (2 of which are in our dataset)
+    compound1 = CompoundFactory()
+    compound_experiment_1A = CompoundExperimentFactory(compound=compound1)
+    compound_experiment_1B = CompoundExperimentFactory(compound=compound1)
+    # Compound 2 only has one experiment (which is in our dataset)
+    compound2 = CompoundFactory()
+    compound_experiment_2 = CompoundExperimentFactory(compound=compound2)
+    # Compound 3 has one experiment (which is not in our dataset)
+    compound3 = CompoundFactory()
+    compound_experiment_3 = CompoundExperimentFactory(compound=compound3)
+
+    matrix = MatrixFactory(
+        [compound_experiment_1A, compound_experiment_1B, compound_experiment_2,],
+        [
```

Comment:
This test is the only code I actually added/changed in this file. The diff is a bit wonky because saving this file auto formatted some of the other code I hadn't touched.

Reply:
Thank you! These tests were really helpful to me in understanding the context here